File: 0006 - ZigZag Conversion/solution.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows == 1) return s;
        vector<string> v(numRows);
        int row = 0, dir = -1;
        for(auto c : s){
            v[row] += c;
            if(row==0 || row==numRows-1) dir = -dir;
            row += dir;
        }
        string res = "";
        for(auto& c : v) res += c;
        return res;
    }
};

File: 0006 - ZigZag Conversion/solution.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows==1:
            return s
        L, row, dir = ['']*numRows, 0, -1
        for c in s:
            L[row] += c
            if row in {0, numRows-1}:
                dir = -dir
            row += dir
        return ''.join(L)
        